webapp/src/RepoGit.ts

```diff
@@ -16,10 +17,30 @@ export class RepoGit {
   private readonly git: IGit;
   private lyraConfig?: LyraConfig;
 
-  constructor(public readonly spConfig: ServerProjectConfig) {
+  constructor(private readonly spConfig: ServerProjectConfig) {
     this.git = new SimpleGitWrapper(spConfig.repoPath);
   }
 
+  public static async cloneIfNotExist(
+    spConfig: ServerProjectConfig,
+  ): Promise<void> {
+    if (!fs.existsSync(spConfig.repoPath)) {
```

Comment:
`existsSync` will block the whole server which could be a problem if the uncerlying filesystem is slow. That is however really unlikely so we decided to replace this call with asynchronous one in a separate merge request.